language-support/scala/bindings-akka/src/main/scala/com/digitalasset/ledger/client/binding/retrying/CommandRetryFlow.scala

```diff
@@ -98,12 +107,18 @@ object CommandRetryFlow {
             Ctx(retryInfo.newRetry, createRetry(retryInfo, failedCompletion), telemetryContext)
         })
 
+        // This buffer, together with the mergePreferred flow, is important to break the cycle around commandSubmission.
+        // Without this setup, mergeRetriesPreferred would backpressure all the way to retryDecider, effectively
+        // stopping the entire flow, i.e. command processing.
+        val buffer =
```

Comment:
So if I understand you correctly we still deadlock if this buffer is full?

Reply:
That's right, but together with the MergePreferred flow this should be dramatically reduced.
Apologies, I should have been explicit about it.